src/openassetio-python/module/hostApi/ManagerBinding.cpp

```diff
@@ -38,5 +44,24 @@ void registerManager(const py::module& mod) {
       .def("createEntityReferenceIfValid", &Manager::createEntityReferenceIfValid,
            py::arg("entityReferenceString"))
       .def("resolve", &Manager::resolve, py::arg("entityReferences"), py::arg("traitSet"),
-           py::arg("context").none(false), py::arg("successCallback"), py::arg("errorCallback"));
+           py::arg("context").none(false), py::arg("successCallback"), py::arg("errorCallback"))
+      .def("preflight", &Manager::preflight, py::arg("entityReferences"), py::arg("traitSet"),
+           py::arg("context").none(false), py::arg("successCallback"), py::arg("errorCallback"))
+      .def(
+          "register",
+          [](Manager& self, const EntityReferences& entityReferences,
+             const trait::TraitsDatas& traitsDatas, const ContextConstPtr& context,
+             const Manager::PublishSuccessCallback& successCallback,
+             const Manager::BatchElementErrorCallback& errorCallback) {
+            // Pybind has no built-in way to assert that a collection
+            // does not contain any `None` elements, so we must add our
+            // own check here.
+            if (std::any_of(traitsDatas.begin(), traitsDatas.end(),
+                            std::logical_not<TraitsDataPtr>{})) {
+              throw pybind11::type_error{"Traits data cannot be None"};
```

Comment:
[question] Based on WIP notes and previous offline chat, I've added this check for no `None` traits datas. Is this OK?

Reply:
Looks spot on, thanks!